src/test/java/no/nav/fo/veilarbdialog/kvp/KvpConsumerTest.java

```diff
@@ -13,39 +22,131 @@
 import org.springframework.kafka.support.SendResult;
 
 import java.time.ZonedDateTime;
+import java.util.UUID;
+
+import static java.time.temporal.ChronoUnit.MILLIS;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static no.nav.fo.veilarbdialog.TestDataBuilder.nyDialog;
+import static no.nav.fo.veilarbdialog.TestDataBuilder.nyHenvendelse;
 
 class KvpConsumerTest extends SpringBootTestBase {
 
     private static final String AKTORID = "4321";
     private static final String SAKSBEHANDLER = "Z99999";
     private static final String BEGRUNNELSE = "Derfor";
+    private static final String AKTOR_ID_1234 = "1234";
     private static final ZonedDateTime AVSLUTTETDATO = ZonedDateTime.now();
+    @Autowired
+    private DialogDAO dialogDAO;
 
     @Autowired
     KafkaTemplate<String, String> producer;
+    Consumer<String, String> kvpAvsluttetConsumer;
 
-    Consumer<String, String> springKvpAvsluttetConsumer;
+    @Value("${application.topic.inn.oppfolgingsperiode}")
+    String oppfolgingsperiodeTopic;
 
     @Value("${application.topic.inn.kvpavsluttet}")
     String kvpAvsluttetTopic;
 
     @BeforeEach
     void setupConsumer() {
-        this.springKvpAvsluttetConsumer = kafkaTestService.createStringStringConsumer(kvpAvsluttetTopic);
+        this.kvpAvsluttetConsumer = kafkaTestService.createStringStringConsumer(kvpAvsluttetTopic);
     }
 
     @SneakyThrows
     @Test
     void behandleKvpAvsluttetConsumerService_spiser_meldinger_fra_kvpAvsluttetTopic() {
-        KvpAvsluttetKafkaDTO kvpAvsluttetKafkaDTO = KvpAvsluttetKafkaDTO.builder()
-                .aktorId(AKTORID)
+        KvpAvsluttetKafkaDTO kvpAvsluttetKafkaDTO = avsluttKvpMelding(AKTORID);
+        SendResult<String, String> sendResult = producer.send(kvpAvsluttetTopic, kvpAvsluttetKafkaDTO.getAktorId(), JsonUtils.toJson(kvpAvsluttetKafkaDTO)).get();
+
+        kafkaTestService.assertErKonsumertAiven(kvpAvsluttetTopic, sendResult.getRecordMetadata().offset(), sendResult.getRecordMetadata().partition(), 10);
+    }
+
+    @SneakyThrows
+    @Test
+    void avslutte_kvp_bruker_som_ikke_er_under_kvp() {
```

Comment:
Passer dette navnet til hva som blir assertet? Ser man sjekker at en dialog ikke er historisk, det burde kanskje være en del av test-navnet da?

Reply:
Gir det egentlig mening at en bruker som ikke er på KVP, avslutter KVP?

Reply:
Jeg syns det gir mening å ha tester for feilsituasjoner som beviser at systemet oppfører seg på en grei måte og ikke skriver feil info til db el.l.